api/src/main/java/jakarta/data/metamodel/ComparableExpression.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.metamodel.constraint.Between;
+import jakarta.data.metamodel.constraint.GreaterThan;
+import jakarta.data.metamodel.constraint.GreaterThanOrEqual;
+import jakarta.data.metamodel.constraint.LessThan;
+import jakarta.data.metamodel.constraint.LessThanOrEqual;
+import jakarta.data.metamodel.constraint.NotBetween;
+import jakarta.data.metamodel.restrict.BasicRestriction;
+
+public interface ComparableExpression<T,V extends Comparable<?>>
+        extends Expression<T, V> {
+
+    // QUESTION: do we really need to expose BasicRestriction here
```

Comment:
No, I can't see a reason to return BasicRestriction rather than Restriction. The only operation for the end user on BasicRestriction is .negate() and that is also on Restriction, so we should return Restriction given that we don't have a need for anything more. That will make autocomplete simpler and more straightforward because it won't be cluttered up with the extra SPI operations that only Jakarta Data providers will be using from BasicRestriction.

Reply:
All done.